src/bin/topiary/main.rs

```diff
@@ -112,7 +113,21 @@ fn run() -> CLIResult<()> {
         unreachable!();
     };
 
-    let mut query = BufReader::new(File::open(query_path)?);
+    let mut query_reader = BufReader::new(File::open(query_path)?);
+
+    // It's not very nice that formatter wants an io::Read and
+    // Configuration::parse wants a &str. Should we just let formatter take
+    // &str as well?
```

Comment:
I quite liked the idea of the formatter being genericised over `io::{Read, Write}` and it makes sense when the configuration parsing is internal. However, yes, if the configuration parsing is factored out, then the API feels a bit clumsy, so unifying it to `&str` seems reasonable.

Reply:
OK, I'll leave that in there as a comment to follow up on separately.